airflow_kubernetes_job_operator/kubernetes_job_operator.py

```diff
@@ -185,8 +204,27 @@ def update_override_params(self, o: dict):
             if isinstance(c, dict):
                 self.update_override_params(c)
 
+    def _validate_job_runner(self):
+        if self.job_runner is not None:
+            return
+        self.prepare_job_runner()
+
+    def prepare_job_runner(self):
+        """Override this method to create your own or augment the job runner"""
+        # create the job runner.
+        self._job_runner: JobRunner = JobRunner(
+            body=self.body,
+            namespace=self.namespace,
+            show_pod_logs=self.get_logs,
+            delete_policy=self.delete_policy,
+            logger=self.logger,
+            auto_load_kube_config=True,
+            name_prefix=self._create_job_name(self.task_id),
+        )
```

Comment:
Unrelated design discussion: wouldn't it be better to return the job runner object here and assign it to `self._job_runner` in the caller, if at all needed?

Reply:
Ah, yes. That is correct. Better to return the value and assign in validation.
- [ ] Fix prepare_job_runner to return a value to the caller.